src/bus/msg.rs

```diff
@@ -0,0 +1,135 @@
+use farcaster_core::{
+    protocol::message::Abort,
+    swap::btcxmr::message::{
+        BuyProcedureSignature, CommitAliceParameters, CommitBobParameters, CoreArbitratingSetup,
+        RefundProcedureSignatures, RevealAliceParameters, RevealBobParameters, RevealProof,
+    },
+    swap::btcxmr::PublicOffer,
+    swap::SwapId,
+};
+use internet2::Api;
+use strict_encoding::{StrictDecode, StrictEncode};
+
+#[derive(Clone, Debug, Display, Api, StrictDecode, StrictEncode)]
+#[api(encoding = "strict")]
+#[display(inner)]
+#[non_exhaustive]
+pub enum Msg {
```

Comment:
given that these are all **protocol** messages, I'd find it ideal if we could find a name that captures this - this name is generic and can be confused with something like `Info` too.
Maybe `Protocol`/`Ptl`/`Prot`/`PtclMsg`/`PrtclMsg`, or fully verbose `ProtocolMsg`.
What's your take on this @h4sh3d?

Reply:
Agree that's not optimal. This matches LNP Node... kinda because in LNP it uses Bolt and Bifrost. A simple one could be `P2pMsg`.

Also in the following PR I'd like to rename `Ctl` into `CtlMsg` and have `BusMsg::Ctl(CtlMsg::...)`, etc. We already have it for `SyncMsg` because `Sync` is too close to some primitive names in Rust language.

- https://github.com/LNP-WG/lnp-node/blob/master/src/bus/mod.rs

Reply:
> A simple one could be P2pMsg.

I'd rather break rust conventions here and do `P2PMsg`/`p2pMsg` 🙈 
Maybe `PeerMsg`.
It would actually be nice if these message would their actual destination or origin - that could make handling easier to follow (something like `FromPeerMsg` or `ToPeerMsg`). What's less nice about that is that this would necessitate converting these types inside peerd. 

> Also in the following PR I'd like to rename Ctl into CtlMsg and have BusMsg::Ctl(CtlMsg::...), etc. We already have it for SyncMsg because Sync is too close to some primitive names in Rust language.

great idea!